components/EditTagsModal.tsx

```diff
@@ -0,0 +1,164 @@
+
+  const initialValues = {
+    tags: collective.tags,
+  };
+
+  const submit = async values => {
+    const { tags } = values;
+    try {
+      const variables = {
+        collective: {
+          id: collective.id,
+          tags: tags,
+        },
+      };
+
+      await editTags({ variables });
+    } catch (e) {
+      addToast({
+        type: TOAST_TYPE.ERROR,
+        message: (
+          <FormattedMessage
+            defaultMessage="Error submiting form: {error}"
+            values={{
+              error: e.message,
+            }}
+          />
+        ),
+      });
+      return;
+    }
+    addToast({
+      type: TOAST_TYPE.SUCCESS,
+      message: <FormattedMessage defaultMessage="Successfully updated tags" />,
+    });
+    handleClose();
+  };
+
+  const handleClose = () => {
+    onClose?.();
+  };
+
+  return (
+    <StyledModal flexGrow={1} maxWidth="500px" onClose={handleClose} trapFocus>
+      <Formik initialValues={initialValues} onSubmit={submit}>
+        {formik => (
+          <Form onSubmit={formik.handleSubmit}>
+            {/* A hack to disable automatic focus from Formik on the tags input.  */}
+            {/* eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex */}
+            <div tabIndex={0} />
+            <ModalHeader onClose={handleClose} hideCloseIcon={true}>
+              <FormattedMessage defaultMessage="Edit tags" />
+            </ModalHeader>
+            <ModalBody mb={0}>
+              <StyledInputFormikField
+                name="tags"
+                htmlFor="tags"
+                labelProps={{ fontWeight: 500, fontSize: '14px', lineHeight: '17px' }}
+              >
+                {({ field }) => {
+                  return (
+                    <CollectiveTagsInput
+                      {...field}
+                      defaultValue={formik.values.tags}
+                      onChange={tags => {
+                        formik.setFieldValue(
+                          'tags',
+                          tags.map(t => t.value.toLowerCase()),
+                        );
+                      }}
+                      suggestedTags={tagStats?.nodes?.map(node => node.tag).filter(tag => !IGNORED_TAGS.includes(tag))}
+                    />
+                  );
+                }}
+              </StyledInputFormikField>
+              <MessageBox type="info" mt={3}>
+                <FormattedMessage
+                  id="collective.tags.info"
+                  defaultMessage="Tags help you improve your group’s discoverability and connect with similar initiatives across the world."
+                />
+              </MessageBox>
+            </ModalBody>
+            <ModalFooter isFullWidth>
+              <Flex justifyContent="flex-start" flexWrap="wrap" gap="16px">
+                <StyledButton
+                  minWidth={132}
+                  buttonStyle="primary"
+                  loading={loading}
+                  disabled={loading || !formik.dirty}
```

Comment:
No need to set `disabled` on loading buttons, clicks are prevented by default

```suggestion
                  disabled={!formik.dirty}
```